apps/api/src/persistence/database/ArangoDatabaseConfiguration.ts

```diff
@@ -0,0 +1,36 @@
+import { PartialDTO } from '../../types/partial-dto';
+
+export default class ArangoDatabaseConfiguration {
+    readonly dbPass: string;
+
+    readonly dbName: string;
+
+    // TODO tighten this type
+    readonly dbHostUrl: string;
+
+    readonly dbRootPass: string;
+
+    readonly dbUser: string;
+
+    constructor({
+        dbPass,
+        dbName,
+        dbHostUrl,
+        dbRootPass,
+        dbUser,
+    }: PartialDTO<ArangoDatabaseConfiguration>) {
+        this.dbPass = dbPass;
+
+        this.dbName = dbName;
+
+        this.dbHostUrl = dbHostUrl;
+
+        this.dbRootPass = dbRootPass;
+
+        this.dbUser = dbUser;
+    }
+
+    public getConfig(): PartialDTO<ArangoDatabaseConfiguration> {
+        return JSON.parse(JSON.stringify(this));
```

Comment:
Just wondering still about that caution about using the JSON method here.

Reply:
As I pointed out before, that article is pretty poorly motivated. `JSON.stringify(...)` will remove methods. This is exactly what you want when converting between an instance and a DTO. You shouldn't be storing symbol-valued props unless you have some elaborate means to persist symbols to your database (we don't).

In fact this "destructive" behaviour is what makes JSON.parse \ stringify work well for this purpose (converting an instance to a DTO).

There is one problem though- I should wrap this logic in a utility so that I can use something like `ClassTransformer` to do this work later if I want and only have to change the logic in one place.

Reply:
Sounds great, yeah the wrapper would be great to have.